client/proposal.go

```diff
@@ -633,6 +642,36 @@ func (c *Client) mpcppParts(
 	return
 }
 
+// secureFundChannel funds the channel and if the funding fails, automatically
+// settles the channel.
+func (c *Client) secureFundChannel(ctx context.Context, ch *Channel, prop ChannelProposal) (*Channel, error) {
+	preFundingTime := time.Now()
+	err := c.fundChannel(ctx, ch, prop)
```

Comment:
The funding deadline of `Funder.Fund` is usually set by the challenge duration. Can you check that this is consistently the case across backends? The mock backend should function similarly. We should revise the Funder mock implementation accordingly.